File: app/src/types/film.ts
export interface Film {
  film_id: number;
  title: string;
  description: string;
  release_year: number;
  language_id: number;
  original_language_id: number | null;
  rental_duration: number;
  rental_rate: number;
  length: number | null;
  replacement_cost: number;
  rating: string;
  special_features: string | null;
  last_update: string;
}

export interface FilmWithCategory extends Film {
  category_name: string;
  category_id: number;
}

export interface FilmWithDetails extends Film {
  language_name: string;
  categories: { category_id: number; name: string }[];
  actors: { actor_id: number; first_name: string; last_name: string }[];
  inventory_count: number;
  available_count: number;
}

export interface FilmFilterParams {
  title?: string;
  categoryId?: number;
  releaseYear?: number;
  rating?: string;
  actorId?: number;
  minLength?: number;
  maxLength?: number;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}

export interface Category {
  category_id: number;
  name: string;
  last_update: string;
}

File: sakila/server/src/main/java/com/sakila/server/model/Film.java
package com.sakila.server.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "film")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Film {

    @Id
    @Column(name = "film_id")
    private Integer filmId;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "release_year")
    private Integer releaseYear;

    @Column(name = "language_id", nullable = false)
    private Integer languageId;

    @Column(name = "original_language_id")
    private Integer originalLanguageId;

    @Column(name = "rental_duration", nullable = false)
    private Integer rentalDuration;

    @Column(name = "rental_rate", nullable = false)
    private BigDecimal rentalRate;

    @Column(name = "length")
    private Integer length;

    @Column(name = "replacement_cost", nullable = false)
    private BigDecimal replacementCost;

    @Column(name = "rating")
    private String rating;

    @Column(name = "special_features")
    private String specialFeatures;

    @Column(name = "last_update", nullable = false)
    private LocalDateTime lastUpdate;
}
